vulture_os/system/cluster/api.py

```diff
@@ -170,6 +171,120 @@ def secret_key(request):
         }, 500)
 
 
+@api_need_key('cluster_api_key')
+@require_http_methods(['GET'])
+def get_message_queues(request):
+    try:
+        params = {}
+        status = request.GET.get('status')
+        if status:
+            if status not in ("new", "running", "done", "failure"):
+                return JsonResponse({
+                    'status': False,
+                    'data': _("Invalid parameter 'status'")
+                }, status=400)
+            params['status'] = status
+        node = request.GET.get('node')
+        if node:
+            params['node__name'] = node
+
+        tasks = MessageQueue.objects.filter(**params).order_by('modified')
+        limit = request.GET.get('limit', "100")
```

Comment:
This get could have been before creating the (final) queryset, instead of creating `tasks` then running `tasks = tasks[:limit]`afterwards to modify the queryset